jwst/resample/resample_spec_step.py

```diff
@@ -24,9 +24,16 @@ class ResampleSpecStep(ResampleStep):
 
     def process(self, input):
         input = datamodels.open(input)
-
+        if isinstance(input, ImageModel):
+            slit_model = datamodels.SlitModel()
+            input.meta.bunit_err = None  # this prevents error array populated in slit model
```

Comment:
Don't reset the value of `bunit_err` in the `input` model. This is causing the BUNIT keyword to get dropped from the header of the ERR extension of the cal file once it's written out at the end of `calwebb_spec2` processing (remember, the cal file doesn't get saved until the very end of the whole pipeline). Instead, reset `bunit_err` in `slit_model` after it's been created and updated (i.e. after line 30).

Reply:
I just push over some code that I think should fix this. The issue is if I don't set bunit_err to
None before I use .update - then I get an error array in the output that I can not get rid of.
Look at what I just pushed over for ImageModel (reample_spec_step) and for Container data - __init__ in resample_spec to see if that will work. I have tested it on MIRI LRs and MIRI_LRS spec3 - which was  added a error array for the miri_lrs_spec3 test - that I missed before

Reply:
You can set both the keyword and the `.err` attribute to None after running `update`.  Or delete them.